doc/rst/language/spec/conversions.rst

```diff
@@ -218,54 +175,235 @@ type ``C`` is allowed when ``D`` is a subclass of ``C``.
+
+     proc g(type t: ParentClass) { }
+     g(owned ChildClass);
+
+
+.. _Implicit_Conversion_Init_Assign:
+
+Implicit Conversions for Initialization and Assignment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An implicit conversion for initialization or assignment occurs at each of
+the following program locations:
+
+-  In an assignment, the expression on the right-hand side of the
+   assignment is converted to the type of the expresssion on the
+   left-hand side of the assignment.
+
+-  In a variable or field declaration that is not a ref variable,
+   the initializing expression is converted to the type of the variable
+   or field. The initializing expression is the right-hand side of the
+   ``=`` in the declaration, if present, or in the field initialization
+   statement in an initializer.
+
+-  The return or yield expression within a function without a ``ref`` or
+   ``const ref`` return intent is converted to the return type of that
+   function.
+
+-  For a call to a function with a formal argument with ``out`` or
+   ``inout`` intent. The value of the formal argument is converted to the
+   type of the corresponding actual argument when setting that actual
+   with assignment or initialization (see :ref:`The_Out_Intent`).
+
+These implicit conversions can be implemented for record types by
+implementing ``init=`` and possibly ``=`` between two types as described in
+:ref:`Advanced_Copy_Initialization` and :ref:`Function_Overloading`.
+``init=`` will be called for initialization as
+described in :ref:`Split_Initialization` and other uses of ``=`` will
+invoke the ``=`` operator.
+
+In the event that an ``=`` overload is provided to support assignment
+between two types, the compiler will check that a corresponding ``init=``
+also exists and emit an error if not.  Additionally, if ``init=`` is
+provided to initialize one type from another, the compiler will check
+that a corresponding ``:`` overload exists and will emit an error if not.
+See also :ref:`Explicit_Conversions` for more information on the ``:``
+operator. It is possible to provide ``:`` without ``init=`` or to provide
+``init=`` without ``=``.
+
+   *Example (implementing-assignment.chpl)*
+
+   Suppose that we have defined a record type to wrap an integer:
+
+   .. code-block:: chapel
+
+      record myInteger {
+        var intValue: int;
+      }
+
+   We might wish to support assignments setting a ``myInteger`` from
+   ``int``. In that event, we can provide the following functions:
+
+   .. code-block:: chapel
+
+      operator =(ref lhs: myInteger, rhs: int) {
+        lhs.intValue = rhs;
+      }
+      proc myInteger.init=(rhs: int) {
+        this.intValue = rhs;
+      }
+      operator :(from: int, type toType: myInteger) {
+        var tmp: myInteger = from; // invoke the init= above
+        return tmp;
+      }
+
+   All three of these functions are required if we wish to support
+   assignment. We can invoke these functions like this:
```

Comment:
"Support assignment" slowed me down in the above. I propose instead:
```
Since we defined `operator =`, including `init=` and `operator :`
is a requirement. [or: are required]
```

Reply:
I wrote
```
   Since we defined ``operator =``, it is necessary to also define 
   ``init=`` and ``operator :`` between these types.
   ```